src/miniscript/decode.rs

```diff
@@ -213,11 +271,11 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> TerminalStack<Pk, Ctx> {
     }
 }
 
-/// Parse a script fragment into an `Terminal`
+/// Parse a script fragment into an `Miniscript`
 #[allow(unreachable_patterns)]
 pub fn parse<Ctx: ScriptContext>(
     tokens: &mut TokenIter,
-) -> Result<Miniscript<bitcoin::PublicKey, Ctx>, Error> {
+) -> Result<Miniscript<Ctx::Key, Ctx>, Error> {
```

Comment:
I am still not getting why Miniscript needs two generic parameters and does not automatically use new `Ctx::Key`

Reply:
i think it's becasue we want to be able to reason about miniscripts with abstract keys that we will later translate, but we only ever want to e.g. parse/compile consensus keys.

Reply:
Correct. Sorry for forgetting